tests/next_tests/unit_tests/program_processor_tests/runners_tests/test_dace_fieldview.py

```diff
@@ -102,35 +102,38 @@ def make_mesh_symbols(mesh: MeshDescriptor):
     )
 
 
-def test_gtir_copy():
+def test_gtir_cast():
     domain = im.call("cartesian_domain")(
         im.call("named_range")(gtir.AxisLiteral(value=IDim.value), 0, "size")
     )
+    IFTYPE_INT64 = ts.FieldType(IFTYPE.dims, dtype=ts.ScalarType(kind=ts.ScalarKind.INT64))
     testee = gtir.Program(
-        id="gtir_copy",
+        id="test_gtir_cast",
         function_definitions=[],
         params=[
-            gtir.Sym(id="x", type=IFTYPE),
+            gtir.Sym(id="x", type=IFTYPE_INT64),
             gtir.Sym(id="y", type=IFTYPE),
             gtir.Sym(id="size", type=SIZE_TYPE),
         ],
         declarations=[],
         body=[
             gtir.SetAt(
-                expr=im.as_fieldop(im.lambda_("a")(im.deref("a")), domain)("x"),
+                expr=im.as_fieldop(
+                    im.lambda_("a")(im.call("cast_")(im.deref("a"), "float64")), domain
+                )("x"),
                 domain=domain,
                 target=gtir.SymRef(id="y"),
             )
         ],
     )
 
-    a = np.random.rand(N)
-    b = np.empty_like(a)
+    a = np.random.randint(-1000, +1000, N)
+    b = np.empty_like(a, dtype=np.float64)
 
     sdfg = dace_backend.build_sdfg_from_gtir(testee, CARTESIAN_OFFSETS)
 
     sdfg(x=a, y=b, **FSYMBOLS)
-    assert np.allclose(a, b)
+    assert np.allclose(a.astype(np.float64), b)
```

Comment:
I think this will not fail, even if you remove conversion.  Probably there needs to be an extra explicit `assert a.astype(np.float64).dtype == b.dtype`

Reply:
Isn't the reason why it does not fail that the integer values are implicitly casted to float, during the comparison? I don't want (and don't need to) verify the type of the output field, because `b.dtype` is defined by the test at line 131.
To provide better test coverage I could apply some operation (e.g. div) on the input array.

Reply:
yeah, my proposal was stupid.
Unfortunately, the current test case still passes without the cast (after removing typing checks on GTIR). The best strategy is probably https://github.com/GridTools/gt4py/blob/646425427d4a608aef1c2ee7bb1a24e3b1e1e3f0/tests/next_tests/integration_tests/feature_tests/iterator_tests/test_builtins.py#L326.